app/lib/theme/widgets/custom/CommunitySourceWidget.tsx

```diff
@@ -0,0 +1,157 @@
+
+  return (
+    <StyledDiv>
+      <Autocomplete
+        readOnly={!!rowId}
+        data-testid="economic-region-autocomplete"
+        onChange={(e, val, reason) => {
+          if (reason === 'clear') {
+            clearWidget();
+          }
+          if (e) {
+            setSelectedEconomicRegion(val);
+          }
+        }}
+        style={{ width: '200px' }}
+        value={selectedEconomicRegion}
+        options={economicRegionOptions}
+        getOptionLabel={(option) => option}
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            data-testid="economic-region-textfield"
+            label="Economic Region"
+          />
+        )}
+      />
+
+      <Autocomplete
+        readOnly={!!rowId}
+        data-testid="regional-district-autocomplete"
+        style={{ width: '200px' }}
+        onChange={(e, val, reason) => {
+          if (reason === 'clear') {
+            setSelectedRegionalDistrict(null);
+            setSelectedGeographicName({ value: null, label: '' });
+          }
+          if (e) {
+            setSelectedRegionalDistrict(val);
+          }
+        }}
+        value={selectedRegionalDistrict}
+        options={
+          regionalDistrictOptions[selectedEconomicRegion]
+            ? [...regionalDistrictOptions[selectedEconomicRegion]]
+            : []
+        }
+        getOptionLabel={(option) => option}
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            data-testid="regional-district-textfield"
+            label="Regional District"
+          />
+        )}
+      />
+
+      <Autocomplete
+        readOnly={!!rowId}
+        data-testid="geographic-name-autocomplete"
+        style={{ width: '200px' }}
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            id="geographic-name-textfield"
+            data-testid="geographic-name-textfield"
+            label="Geographic Name"
+          />
+        )}
+        options={
+          geographicNameOptions[selectedRegionalDistrict]
+            ? [...geographicNameOptions[selectedRegionalDistrict]]
+            : []
+        }
+        isOptionEqualToValue={(option, val) => {
+          return option.value === val.value;
+        }}
+        getOptionLabel={(option) => {
+          return option.label ?? '';
+        }}
+        value={selectedGeographicName}
+        onChange={(e, val, reason) => {
+          if (reason === 'clear') {
+            setSelectedGeographicName({ value: null, label: '' });
+            return;
+          }
+          if (e) {
+            setSelectedGeographicName(val);
+            onChange({
+              bcGeographicName: val.label,
+              economicRegion: selectedEconomicRegion,
+              regionalDistrict: selectedRegionalDistrict,
+              geographicNameId: val.value,
+            });
+          }
+        }}
+      />
+    </StyledDiv>
+  );
+};
```

Comment:
a small thing, may be a little room before buttons, 
![image](https://github.com/user-attachments/assets/17808bba-efcf-46d1-81bd-90273c930c38)

Reply:
Should be fixed now